src/nativeComponents.ts

```diff
@@ -0,0 +1,118 @@
+import fs from 'fs';
+import stream from 'stream';
+
+import elfy from 'elfy';
+import lodash from 'lodash';
+import PluginError from 'plugin-error';
+import Vinyl from 'vinyl';
+
+const PLUGIN_NAME = 'nativeComponents';
+
+function formatUUID(uuid: string) {
```

Comment:
`formatUUID` is used exactly once, to convert a UUID in binary form to a canonical-form string. Change it to take a Buffer?
```suggestion
function formatUUID(buffer: Buffer & { length: 8 }) {
  const uuid = buffer.toString('hex');
```

Reply:
Couldn't find a nice way to prove the length type to the compiler, so took a slightly different approach.